### PR TITLE
Reduce complexity in photos controller

### PR DESCRIPTION
**Pull request by Br3nda**
according to rubocop this method had the highest perceived complexity 
Moved some to another method.

### CODE REVIEW

**Comment by Br3nda** [comment]
```
 1) account types admin user deleting account type
     Failure/Error: visit account_type_path account_type
     Capybara::Poltergeist::StatusFailError:
       Request to 'http://localhost:8081/account_types/2' failed to reach server, check DNS and/or server status
     # ./spec/features/admin/account_types_spec.rb:52:in `block (3 levels) in <top (required)>'
```

*sigh*.. again.

File: app/controllers/photos_controller.rb [hunk 1]
```diff
@@ -129,4 +115,23 @@ def photo_params
...
+  def find_or_create_from_flickr_photo(flickr_photo_id)
+    photo = Photo.find_by(flickr_photo_id: flickr_photo_id)
+    photo = Photo.new(photo_params) unless photo
+    photo.owner_id = current_member.id
+    photo.set_flickr_metadata
+    photo
+  end
+
+  def which_collection?
+    case params[:type]
```
Comment (CloCkWeRX) [inline app/controllers/photos_controller.rb]: Should really swap this to something like
```
return unless %w(garden planting harvest).index(planting[:type])

@photo.send(planting[:type].plualize.to_sym)
```

**Comment by Br3nda** [comment]
code climate is complaining of code duplication in growstuff.rake. 
So i fixed that too...

File: app/controllers/photos_controller.rb (hunk 1 above)
Comment (Br3nda) [inline app/controllers/photos_controller.rb]: :+1: will do

Comment (Br3nda) [inline app/controllers/photos_controller.rb]: neat! same list is used when finding item later.

**Comment by Br3nda** [comment]
Brakeman is unhappy with our 2 liners, and brakeman is right.
I'll make it the more verbose, but stronger white-listed way.

**Pull request merged**